### PR TITLE
Backend/BugFix/CustomGroupOrder

### LINKED ISSUE #127: Redesign custom group order
Sort order Double type is not good for fetching data, must be changed to Integer

### PR DESCRIPTION
**Pull request by PavloLuchyk**
[127][BE]Changed Group sortOrder to Interger

SQL:
alter table groups alter column sorting_order type integer using sorting_order::integer;

### CODE REVIEW

File: src/main/java/com/softserve/repository/impl/GroupRepositoryImpl.java [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 import com.softserve.entity.Group;
 import com.softserve.repository.GroupRepository;
+import io.swagger.models.auth.In;
```
Comment (yuremboo) [inline src/main/java/com/softserve/repository/impl/GroupRepositoryImpl.java]: I don't think it should be here

File: src/main/java/com/softserve/repository/impl/GroupRepositoryImpl.java [hunk 2]
```diff
@@ -64,7 +65,7 @@
     private static final String UPDATE_GROUP_OFFSET
             = "UPDATE Group g "
             + "SET g.sortingOrder = g.sortingOrder+1 "
-            + "WHERE g.sortingOrder >= :position";
+            + "WHERE g.sortingOrder >= :lowerPosition and g.sortingOrder < :upperPosition";
```
Comment (yuremboo) [inline src/main/java/com/softserve/repository/impl/GroupRepositoryImpl.java:67]: why do you need this statement? and g.sortingOrder < :upperPosition"

File: src/main/java/com/softserve/service/impl/GroupServiceImpl.java [hunk 3]
```diff
@@ -127,23 +128,20 @@ public Group updateGroupOrder(Group group, Long afterId) {
...
-            Double previousPosition = getSortingOrderById(afterId);
-            Double nextPosition = groupRepository.getNextPosition(previousPosition)
-                    .orElse(previousPosition+2);
-            Double newPosition = (previousPosition+nextPosition)/2;
-            if (newPosition - 1 < 0.01) {
-                newPosition = groupRepository.getMaxSortingOrder().orElse(0.0)+1;
-            }
-            group.setSortingOrder(newPosition);
+            Integer lowerBound = getSortingOrderById(afterId)+1;
+            Integer upperBound = Optional.ofNullable(group.getSortingOrder()).orElse(maxOrder+1)+1;
```
Comment (yuremboo) [inline src/main/java/com/softserve/service/impl/GroupServiceImpl.java:136]: Didn't get why do you need 'upperBound'

**Review by yuremboo — COMMENTED** [review]
Fine, but please clarify regarding upperPosition

File: src/main/java/com/softserve/repository/impl/GroupRepositoryImpl.java (hunk 2 above)
Comment (PavloLuchyk) [reply src/main/java/com/softserve/repository/impl/GroupRepositoryImpl.java:67]: It is used for update, if we want to update sort order of some group by setting its order at the beginning, we do not need to change offset of the elements that come after group, so we change only in specific interval

File: src/main/java/com/softserve/service/impl/GroupServiceImpl.java (hunk 3 above)
Comment (PavloLuchyk) [reply src/main/java/com/softserve/service/impl/GroupServiceImpl.java:134]: It is used for update, to change sort orders only in specific interval

Comment (PavloLuchyk) [reply src/main/java/com/softserve/service/impl/GroupServiceImpl.java:134]: > It is used for update, to change sort orders only in specific interval

Comment (PavloLuchyk) [reply src/main/java/com/softserve/service/impl/GroupServiceImpl.java:136]: It is used for update, to change sort orders only in specific interval

File: src/main/java/com/softserve/repository/impl/GroupRepositoryImpl.java (hunk 1 above)
Comment (PavloLuchyk) [reply src/main/java/com/softserve/repository/impl/GroupRepositoryImpl.java]: Removed

**Pull request merged**